tests/bluetooth/host/keys/bt_keys_add_type/src/test_suite_add_type_invalid_inputs.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <bluetooth/bluetooth.h>
+#include <keys.h>
+#include "host_mocks/assert.h"
+#include "mocks/keys_help_utils.h"
+
+/*
+ *  Test function with NULL key reference
+ *
+ *  Constraints:
+ *   - Any key type can be used
+ *   - Key reference is passed as NULL
+ *
+ *  Expected behaviour:
+ *   - An assertion is raised and execution stops
+ */
+void test_null_key_reference(void)
+{
+	expect_assert();
+	bt_keys_add_type(NULL, 0x00);
```

Comment:
If this bt_keys_add_type is a fake, this is not testing the NULL case. If it's not a fake, I would expect this to actually result in a segmentation fault or protection violation. Has this been executed?

I might recommend that you consider adding an assertion to bt_keys_add_type() for an invalid pointer, though I don't know if that's in scope for this project.

Reply:
Also, is a valid key_ref and zero mask tested? It appears to have no effect, which may become a relied-upon behavior.

Reply:
> I might recommend that you consider adding an assertion to bt_keys_add_type() for an invalid pointer, though I don't know if that's in scope for this project.

Actually, this is a test for the assert to be triggered (which I added).

> Also, is a valid key_ref and zero mask tested? It appears to have no effect, which may become a relied-upon behavior.

I didn't think about valid key with zero mask, but it seems to be a valid case

Reply:
I'm not clear on why an assert would be trigged. What triggers it? Keep in mind, if the original code is running, the current implementation is:

`keys->keys |= type;`

If the keys pointer is null, this is accessing memory at an offset from 0x0:

`*(uint16_t *)(((uint8_t *)NULL)+offsetof(struct bt_keys, keys)) = new_val;`

That's not going to assert, that's going to hard-fault (on most platforms).

Reply:
Sorry, may be I wasn't clear enough.

I already modified the keys.c a little bit to include such asserts and you can find the modifications here [Enhance bluetooth host keys](https://github.com/zephyrproject-rtos/zephyr/pull/49596).

The UTs projects are dependent on these modifications and that's why CI checks are always failing because the modifications need to be merged first.